quartz/depgraph.ts

```diff
@@ -0,0 +1,123 @@
+import Graph from "graphology"
+
+export default class DepGraph {
+  graph: Graph
+
+  constructor() {
+    this.graph = new Graph({ allowSelfLoops: false, multi: false, type: "directed" })
+  }
+
+  toString(): string {
+    return JSON.stringify(
+      this.graph.toJSON()["edges"].map((obj) => obj["key"]),
+      null,
+      2,
+    )
+  }
+
+  // For the node provided:
+  // If node does not exist, add it
+  // If an edge was added in other, it is added in this graph
+  // If an edge was deleted in other, it is deleted in this graph
+  mergeEdgesForNode(other: DepGraph, node: string): void {
```

Comment:
this adds quite a bit of complexity, can't we just use one depgraph for everything?

Reply:
Commented in this [thread](https://github.com/jackyzha0/quartz/pull/716/files#r1472469272)